tools/wpt/browser.py

```diff
@@ -939,44 +944,214 @@ class Chrome(ChromeChromiumBase):
+        if len(version_parts) < 3:
+            self.logger.info(f"Version {version} could not be formatted for build matching.")
+            return None
+        return ".".join(version_parts[0:3])
 
-        version = self._remove_version_suffix(version)
+    def _get_webdriver_url(self, version, channel):
+        """Get a ChromeDriver URL to download a version of ChromeDriver that matches
+        the browser binary version.
+        """
+        # Remove version suffix if it exists.
+        if version:
+            version = self._remove_version_suffix(version)
 
-        parts = version.split(".")
-        assert len(parts) == 4
-        latest_url = ("https://chromedriver.storage.googleapis.com/LATEST_RELEASE_"
-                      f"{'.'.join(parts[:-1])}")
+        # Check if a file exists containing the matching ChromeDriver version download URL.
+        # This is generated when installing Chrome for Testing using the install command.
+        download_url_reference_file = os.path.join(
+            self._get_browser_binary_dir(None, channel), "matching_chromedriver_url.txt")
+        if os.path.isfile(download_url_reference_file):
+            self.logger.info("Download info for matching ChromeDriver version found.")
+            with open(download_url_reference_file, "r") as f:
+                return f.read()
+
+        # If no ChromeDriver download URL reference file exists,
+        # try to find a download URL based on the build version.
+        self.logger.info(f"Searching for ChromeDriver downloads for version {version}.")
+        return self._get_webdriver_url_by_build(version)
+
+
+    def _get_webdriver_url_by_build(self, version):
+        """Find a ChromeDriver download URL based on a MAJOR.MINOR.BUILD version.
+
+        Returns: Download URL string or None if no matching build is found.
+        """
+        version = self._get_build_version(version)
         try:
-            latest = get(latest_url).text.strip()
-        except requests.RequestException:
-            latest_url = f"https://chromedriver.storage.googleapis.com/LATEST_RELEASE_{parts[0]}"
-            try:
-                latest = get(latest_url).text.strip()
-            except requests.RequestException:
-                # We currently use the latest Chromium revision to get a compatible Chromedriver
-                # version for Chrome Dev, since it is not available through the ChromeDriver API.
-                # If we've gotten to this point, it is assumed that this is Chrome Dev.
-                filename = f"chromedriver_{self._chromedriver_platform_string}.zip"
-                revision = self._get_chromium_revision(filename, version)
-                return self._build_snapshots_url(revision, filename)
-        return f"https://chromedriver.storage.googleapis.com/{latest}/{filename}"
+            # Get a list of builds with download URLs from Chrome for Testing.
+            resp = get(
+                "https://googlechromelabs.github.io"
```

Comment:
Now that the url doesn't depend on input, this can be moved to a constant at the top of the file now.